### ISSUE TITLE
One or more environment variables failed assertions: SPAPI_ENDPOINT is missing.

### ISSUE THREAD

**Issue by Nilanth** [jlevers/selling-partner-api#18]
Getting **One or more environment variables failed assertions: SPAPI_ENDPOINT is missing**. error if it is not mentioned in env, but it is set using dynamic config.

**Comment by cpiggott** [comment]
Are you setting the environment variables dynamically, or are you constructing the configuration dynamically?

**Comment by cpiggott** [comment]
Can you post an example of your code?

**Comment by jlevers** [comment]
@Nilanth try something like this:

```php
$config = new SellingPartnerApi\Configuration([/* auth config array */], 'https://your-endpoint.amazon.com');
$api = new SellingPartnerApi\Api\ReportsApi($config);
```

**Comment by cpiggott** [comment]
@jlevers 

This won't work. The constructor `new SellingPartnerApi\Configuration()` immediately checks to see if the required environment variables are present. The `allVarsLoaded()` is going to return false, and then attempt to load in the environment variables using the `loadDotEnv()` in the `env.php` file. 



If someones plans to pass in a configuration as opposed to using ENV variables, there probably needs to be a check for the `$lwaAuthInfo` and not look for environment variables, and instead validate the configuration passed to the constructor. If that validation fails, then the constructor should return false and a ValidationException should be thrown.
 
 I can start working on this if you'd like me to. Happy to contribute.

**Comment by jlevers** [comment]
Sorry, I glossed over the part of the OP where @Nilanth says that they'd like to not include the endpoint in the `.env`, so you're absolutely correct that it won't work. This is related to #1, which I haven't gotten to yet.

If you want to work on a PR related to this, that would be awesome! Feel free to email me if you need help/clarification with anything: `jesse [at] jesse${lastname}.com`

**Comment by jlevers** [comment]
This was fixed by #19.
